### ISSUE TITLE
Add aria-label to icon buttons

### ISSUE THREAD

**Issue by pngwn** [gradio-app/gradio#2777]
### Describe the bug

cc @emilyuhde

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

-

### Logs

```shell
-
```

### System Info

```shell
-
```

### Severity

serious, but I can work around it

**Comment by emilyuhde** [comment]
Repro steps:

1. Go to the doc site for the Image component: https://gradio.app/docs/#image
2. Upload an image of your choice
3. Inspect the edit and close button in the button right hand corner of the component

Expected: There should be an aria-label on these button elements to give screen reader users labels describing their functionality
Actual: No aria-labels or an equivalent (aria-labelledby, etc) is present

Alternate repro method:

1. Turn on screenreader of your choice (I used Voiceover)
2. Tab down to the button elements described in step 3 above

You'll notice the screen reader reads out "Button, group" with no label.

<img width="1448" alt="Screen Shot 2022-12-08 at 8 54 24 PM" src="https://user-images.githubusercontent.com/4451556/206605543-9e70f7ac-c0f3-4d13-9bae-18ee89950b19.png">
